.github/workflows/rust-bindings.yml

```diff
@@ -0,0 +1,39 @@
+---
+name: Rust Bindings
+
+on:
+  push:
+    paths:
+      - "include/"
+      - "src/"
+      - "rust/"
+      - "*akefile*"
+    branches:
+      - main
+  pull_request:
+
+jobs:
+  cargo-test:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+      - name: Set up Ruby
+        uses: ruby/setup-ruby@v1
+        with:
+          ruby-version: head
+          bundler-cache: true
+      - name: Bundle install
+        run: bundle install
+      - uses: actions/cache@v3
+        with:
+          path: |
+            ~/.cargo/registry
+            ~/.cargo/git
+            target
+          key: ${{ runner.os }}-cargo-${{ hashFiles('**/Cargo.toml') }}
+          restore-keys: |
+            ${{ runner.os }}-cargo-${{ hashFiles('Cargo.toml') }}
+            ${{ runner.os }}-cargo
+      - name: cargo test
```

Comment:
GHA workflows for rust crates typically use a matrix strategy and test against multiple rust versions. I see 1.59.0 mentioned as the MSRV in the thread. Testing against that and stable seems useful.

As it is, this workflow tests against whatever version of the rust toolchain happens to be baked into the GHA runner image.

https://github.com/dtolnay/rust-toolchain is a popular action used to setup a Rust toolchain. I use the collection of workflows in https://github.com/artichoke/setup-rust for Artichoke.

Reply:
> GHA workflows for rust crates typically use a matrix strategy and test against multiple rust versions.

I understand that this is common, but is it _really_ typical? I guess IMO, I'd rather just test against current stable (or rather what's on the GH agent) until there's a real need to add the build complexity. And if that means only claiming "we only support Rust version [the latest stable]," then I'm all for that since that gives development flexibility until we decide we want to lock it down. On the other hand, I've only ever done private/company-internal releases, so I don't have OSS+Rust experience and may be off here. I just like the idea of pushing out complexity until it's needed.

> https://github.com/dtolnay/rust-toolchain is a popular action used to setup a Rust toolchain. 

I've never used that; only https://github.com/actions-rs, but whatever's clever. Again, I'd rather keep it simple until there's a need. If anyone else feels strongly about matrix testing, please chime in though and I'll update.

Reply:
The actions-rs organization and all of its repositories are widely understood to be abandoned: https://github.com/actions-rs/toolchain/issues/216

Reply:
> The actions-rs organization and all of its repositories are widely understood to be abandoned: [actions-rs/toolchain#216](https://github.com/actions-rs/toolchain/issues/216)

I guess not widely enough. :D

Reply:
😅 sorry about that. the @actions-rs actions have been emitting warnings to the workflow page for upwards of a year due to not being updated for several GitHub Actions security advisories and the pending node 12 runtime removal.

Reply:
haha no worries! Somehow I've managed not to see those warnings. We've used them at work for quite some time for most of our non-trivial Rust projects; I'll have to check those tomorrow now!